oauth2/src/main/java/com/okta/spring/oauth/http/UserAgent.java

```diff
@@ -0,0 +1,485 @@
+    private static final String OKTA_SDK_SERVLET_ID = "okta-servlet-java";
+    private static final String OKTA_SDK_SERVLET_CLASS = "com.okta.sdk.servlet.config.Config";
+
+    private static final String OKTA_SDK_RUNTIME_SPRING_WEBMVC_ID = "okta-spring-webmvc";
+    private static final String OKTA_SDK_RUNTIME_SPRING_WEBMVC_CLASS = "com.okta.spring.mvc.AbstractSpringControllerConfig";
+
+    //Okta Spring Boot
+    private static final String OKTA_SDK_SPRING_BOOT_STARTER_ID = "okta-spring-boot-starter";
+    private static final String OKTA_SDK_SPRING_BOOT_STARTER_CLASS = "com.okta.spring.boot.autoconfigure.OktaAutoConfiguration";
+
+    //Integration Runtimes
+    private static final String INTEGRATION_RUNTIME_SPRING_ID = "spring";
+    private static final String INTEGRATION_RUNTIME_SPRING_CLASS = "org.springframework.context.ApplicationContext";
+
+    //Rapid Prototyping
+    private static final String RAPID_PROTOTYPING_SPRING_BOOT_ID = "spring-boot";
+    private static final String RAPID_PROTOTYPING_SPRING_BOOT_CLASS = "org.springframework.boot.SpringApplication";
+
+    //Runtime
+    private static final String JAVA_SDK_RUNTIME_STRING = "java";
+
+    ////Additional Information////
+
+    //Security Frameworks
+    private static final String SECURITY_FRAMEWORK_SHIRO_ID = "shiro";
+    private static final String SECURITY_FRAMEWORK_SHIRO_CLASS = "org.apache.shiro.SecurityUtils";
+    private static final String SECURITY_FRAMEWORK_SPRING_SECURITY_ID = "spring-security";
+    private static final String SECURITY_FRAMEWORK_SPRING_SECURITY_CLASS = "org.springframework.security.core.SpringSecurityCoreVersion";
+
+    //Web Servers
+    private static final String WEB_SERVER_TOMCAT_ID = "tomcat";
+    private static final String WEB_SERVER_TOMCAT_BOOTSTRAP_CLASS = "org.apache.catalina.startup.Bootstrap";
+    private static final String WEB_SERVER_TOMCAT_EMBEDDED_CLASS = "org.apache.catalina.startup.Tomcat";
+    private static final String WEB_SERVER_JETTY_ID = "jetty";
+    private static final String WEB_SERVER_JETTY_CLASS = "org.eclipse.jetty.servlet.listener.ELContextCleaner";
+    private static final String WEB_SERVER_JBOSS_ID = "jboss";
+    private static final String WEB_SERVER_JBOSS_CLASS = "org.jboss.as.security.plugins.AuthenticationCacheEvictionListener";
+    private static final String WEB_SERVER_WEBSPHERE_ID = "websphere";
+    private static final String WEB_SERVER_WEBSPHERE_CLASS = "com.ibm.websphere.product.VersionInfo";
+    private static final String WEB_SERVER_GLASSFISH_ID = "glassfish";
+    private static final String WEB_SERVER_GLASSFISH_CLASS = "com.sun.enterprise.glassfish.bootstrap.GlassFishMain";
+    private static final String WEB_SERVER_WEBLOGIC_ID = "weblogic";
+    private static final String WEB_SERVER_WEBLOGIC_CLASS = "weblogic.version";
+    private static final String WEB_SERVER_WILDFLY_ID = "wildfly";
+    private static final String WEB_SERVER_WILDFLY_CLASS = "org.jboss.as.security.ModuleName";
```

Comment:
There are a lot of possibilities for user agent values here. Do we want/need to capture all of these?

Reply:
yeah, copied from the SDK